core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -94,20 +86,27 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
         m_result_ptr(reducer.view().data()) {}
 
  private:
-  template <typename TagType>
-  std::enable_if_t<std::is_void<TagType>::value> exec(reference_type update) {
-    using member_type = typename Policy::member_type;
-    member_type e     = m_policy.end();
-    for (member_type i = m_policy.begin(); i < e; ++i) m_functor(i, update);
-  }
+  template <typename T>
+  struct HasJoin {
```

Comment:
I had (more) trouble with the current detection mechanism and only require a specific signature here. More concretely, the current implementation doesn't work if there are multiple overloads.

Reply:
Did you consider updating `ReduceFunctorHasJoin`?

Reply:
I can't do that since `ReduceFunctorHasJoin` detects if there is any overload. It doesn't check for a specific signature.
Since we expect a certain signature anyway I could try to see if asking for a specific signature in all the places where `ReduceFunctorHasJoin` is used instead would work. I would prefer a different pull request for that, though.